boto/kinesis/layer1.py

```diff
@@ -591,66 +653,171 @@ def put_record(self, stream_name, data, partition_key,
+        data records with the same partition key map to the same shard
+        within the stream. For more information, see `Partition Key`_
+        in the Amazon Kinesis Developer Guide .
+
+        Each record in the `Records` array may include an optional
+        parameter, `ExplicitHashKey`, which overrides the partition
+        key to shard mapping. This parameter allows a data producer to
+        determine explicitly the shard where the record is stored. For
+        more information, see `Adding Multiple Records with
+        PutRecords`_ in the Amazon Kinesis Developer Guide .
+
+        The `PutRecords` response includes an array of response
+        `Records`. Each record in the response array directly
+        correlates with a record in the request array using natural
+        ordering, from the top to the bottom of the request and
+        response. The response `Records` array always includes the
+        same number of records as the request array.
+
+        The response `Records` array includes both successfully and
+        unsuccessfully processed records. Amazon Kinesis attempts to
+        process all records in each `PutRecords` request. A single
+        record failure does not stop the processing of subsequent
+        records.
+
+        A successfully-processed record includes `ShardId` and
+        `SequenceNumber` values. The `ShardId` parameter identifies
+        the shard in the stream where the record is stored. The
+        `SequenceNumber` parameter is an identifier assigned to the
+        put record, unique to all records in the stream.
+
+        An unsuccessfully-processed record includes `ErrorCode` and
+        `ErrorMessage` values. `ErrorCode` reflects the type of error
+        and can be one of the following values:
+        `ProvisionedThroughputExceededException` or `InternalFailure`.
+        `ErrorMessage` provides more detailed information about the
+        `ProvisionedThroughputExceededException` exception including
+        the account ID, stream name, and shard ID of the record that
+        was throttled.
+
+        Data records are accessible for only 24 hours from the time
+        that they are added to an Amazon Kinesis stream.
+
+        :type records: list
+        :param records: The records associated with the request.
+
+        :type stream_name: string
+        :param stream_name: The stream name associated with the request.
+
+        :type b64_encode: boolean
+        :param b64_encode: Whether to Base64 encode `data`. Can be set to
+            ``False`` if `data` is already encoded to prevent double encoding.
+
+        """
+        params = {'Records': records, 'StreamName': stream_name, }
+        if b64_encode:
+            for i in range(len(params['Records'])):
+                data = params['Records'][i]['Data']
+                if not isinstance(data, six.binary_type):
+                    params['Records'][i]['Data'] = data.encode('utf-8')
```

Comment:
Should this be `data = data.encode('utf-8')`? Otherwise we are never using the encoded value.